src/Compilers/CSharp/Portable/Symbols/LocalDeclarationKind.cs

```diff
@@ -49,5 +49,16 @@ internal enum LocalDeclarationKind : byte
         /// The variable that captures the result of a pattern matching operation like "i" in "expr is int i"
         /// </summary>
         PatternVariable,
+
+        /// <summary>
+        /// User variable declared by a declaration expression in the left-hand-side of a deconstruction assignment.
+        /// </summary>
+        Deconstruction,
+
+        /// <summary>
+        /// User variable declared by a declaration expression that is not a deconstruction. This occurs for an
+        /// out variable declaration, and as a result of error recovery in incorrect code.
+        /// </summary>
+        DeclarationExpression,
```

Comment:
This is precisely the precision that the compiler uses internally to distinguish among the kinds of declarations that it treats differently. The purpose of this member is to expose rather than discard the information we already have by virtue of the control-flow within the compiler when these symbols are created, and it reflects the categorization that we have already found useful within the compiler.